### PR TITLE
Fix recovering all on deleted

### LINKED ISSUE #534: `multipass recover foo` recovers all deleted instances if `foo` is not deleted
```sh
$ multipass list
Name                    State             IPv4             Release
pliable-greyhound       STOPPED           --               Ubuntu 16.04 LTS
definite-shark          DELETED           --               Not Available
$ multipass recover pliable-greyhound
$ multipass list
Name                    State             IPv4             Release
definite-shark          STOPPED           --               Ubuntu 16.04 LTS
pliable-greyhound       STOPPED           --               Ubuntu 16.04 LTS
```

I'm afraid this is my fault: I inadvertently introduced this with #528 

Fix coming shortly.

### PR DESCRIPTION
**Pull request by ricab**
Fixes #534 
This can only be merged after #535 !

### CODE REVIEW

File: src/daemon/daemon.cpp [hunk 1]
```diff
@@ -1277,39 +1270,32 @@ try // clang-format on
...
-        if (it == deleted_instances.end())
+        for (const auto& name : instances)
         {
-            it = vm_instances.find(name);
-            if (it == vm_instances.end()) // noop otherwise
-                fmt::format_to(errors, "instance \"{}\" does not exist\n", name);
+            auto it = deleted_instances.find(name);
+            if(it != std::end(deleted_instances))
+            {
+                it->second->shutdown(); // not sure why this is needed, but leaving it alone
```
Comment (townsend2010) [inline src/daemon/daemon.cpp]: I imagine this is a copy & pasta error from back when:)  I think it's safe to remove.

File: src/daemon/daemon.cpp [hunk 2]
```diff
@@ -1910,6 +1898,28 @@ void mp::Daemon::start_mount(const VirtualMachine::UPtr& vm, const std::string&
                      Qt::QueuedConnection);
 }
 
+std::string mp::Daemon::check_instance_alive(const std::string& instance_name) const
```
Comment (townsend2010) [inline src/daemon/daemon.cpp]: IMO, the name of this method implies that we are checking that the instance is up and running.  I think maybe a better name would be something like `check_instance_usable` or `check_instance_startable`.

File: src/daemon/daemon.cpp (hunk 1 above)
Comment (ricab) [reply src/daemon/daemon.cpp]: Yes it looks like that. Alright, I will remove it.

File: src/daemon/daemon.cpp (hunk 2 above)
Comment (ricab) [reply src/daemon/daemon.cpp]: You're right, I was struggling with naming there... From your suggestions I prefer `usable`. I am still not sure it conveys the meaning we want (not deleted), but it's certainly usable :stuck_out_tongue: 

Ahh I am sure there is an English word for it, I just can't seem to find it...

Comment (townsend2010) [reply src/daemon/daemon.cpp]: Lets go with `operational`:)

Comment (ricab) [reply src/daemon/daemon.cpp]: OK, sounds good, will do that.

**Pull request merged**